Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfUtility.cs

```diff
@@ -0,0 +1,370 @@
+                    //{
+                    //    for (int j = 0; j < gltfObject.materials.Length; j++)
+                    //    {
+                    //        if (!string.IsNullOrEmpty(gltfObject.materials[i].name) &&
+                    //            gltfObject.materials[i].name == extensionRequired.Key)
+                    //        {
+                    //            gltfObject.materials[i].Extensions.Add(gltfObject.extensionsUsed[i], extensionRequired.Value);
+                    //            var extension = JsonUtility.FromJson<KHR_Materials_PbrSpecularGlossiness>(extensionRequired.Value);
+                    //            extension.ElementName = gltfObject.materials[i].name;
+                    //            gltfObject.RegisteredExtensions.Add(extension);
+                    //        }
+                    //    }
+                    //}
+                    //else
+                    {
+                        Debug.LogWarning($"Unsupported Extension: {gltfObject.extensionsRequired[i]}");
+                        return null;
+                    }
+                }
+            }
+
+            for (int i = 0; i < gltfObject.extensionsUsed?.Length; i++)
+            {
+                var extensionsUsed = GetGltfExtensionObjects(jsonString, gltfObject.extensionsUsed[i]);
+
+                foreach (var extensionUsed in extensionsUsed)
+                {
+                    // TODO Update this after KHR_materials_pbrSpecularGlossiness extension is supported
+                    //if (gltfObject.extensionsUsed[i].Equals("KHR_materials_pbrSpecularGlossiness"))
+                    //{
+                    //    for (int j = 0; j < gltfObject.materials.Length; j++)
+                    //    {
+                    //        if (!string.IsNullOrEmpty(gltfObject.materials[i].name) &&
+                    //            gltfObject.materials[i].name == extensionUsed.Key)
+                    //        {
+                    //            gltfObject.materials[i].Extensions.Add(gltfObject.extensionsUsed[i], extensionUsed.Value);
+                    //            var extension = JsonUtility.FromJson<KHR_Materials_PbrSpecularGlossiness>(extensionUsed.Value);
+                    //            extension.ElementName = gltfObject.materials[i].name;
+                    //            gltfObject.RegisteredExtensions.Add(extension);
+                    //        }
+                    //    }
+                    //}
+                    //else
+                    {
+                        Debug.LogWarning($"Unsupported Extension: {gltfObject.extensionsUsed[i]}");
+                    }
+                }
+            }
+
+            var meshPrimitiveAttributes = GetGltfMeshPrimitiveAttributes(jsonString);
+
+            for (int i = 0; i < gltfObject.meshes.Length; i++)
+            {
+                for (int j = 0; j < gltfObject.meshes[i].primitives.Length; j++)
```

Comment:
we should have some sort of check that we never try and access an item in the meshPrimitiveAttributes that doesn't exist. Even though these items are based on the same logic, some sort of regression could occur where the meshPrimitiveAttributes size isn't equal to the sum of the mesh primitive lengths.

Reply:
picking up this fix